radix.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+import math
+import string
+
+
+def radix_integer(lst):
+    """
+    Return a sorted version of a list of integers using radix sort
+    """
+    for i in lst:
+        assert isinstance(i, int), "Not all integers"
+
+    if len(lst) < 1:
+        return lst
+
+    exp = int(math.log10(max(lst))) + 1  # calculate largest exponent needed
+    for i in range(exp):
+        buckets = [[] for j in range(10)]
+        for x in lst:
+            buckets[x % 10 ** (i + 1) // 10 ** i].append(x)
+        lst = reduce(lambda x, y: x + y, buckets)   # concatenate all buckets
+    return lst
+
+
+def radix_string(s):
+    for c in s:
+        assert isinstance(c, (str, unicode)), "Not all strings"
+
+    if len(s) < 1:
+        return s
+
+    max_string = max([len(x) for x in s])
+
+    # create custom ord dictionary based on standard printable characters
+    # let order and priority be the default of the library module
+    ordinal = dict(zip(string.printable, range(1, len(string.printable) + 1)))
+    ordinal[''] = 0   # for after end of words
+
+    for i in range(max_string - 1, -1, -1):   # use least sig digit first
+        buckets = [[] for j in range(len(ordinal))]
+        for item in s:
+            c = item[i: i + 1]
+            buckets[ordinal[c]].append(item)
+        s = reduce(lambda x, y: x + y, buckets)
```

Comment:
also look at `itertools` (just because it's fun)

```python
In [68]: stuff = [['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h']]

In [69]: stuff
Out[69]: [['a', 'b', 'c'], ['d', 'e', 'f'], ['g', 'h']]

In [70]: import itertools

In [71]: [l for l in itertools.chain(*stuff)]
Out[71]: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
```